src/generate/register.rs

```diff
@@ -213,24 +213,25 @@ pub fn fields(
             } = f.bit_range;
             let sc = f.name.to_sanitized_snake_case();
             let pc = f.name.to_sanitized_upper_case();
-            let pc_r = Ident::from(&*format!("{}_A", pc));
```

Comment:
Why all the `from` -> `new` changes? I've introduced the `from` so the update wouldn't require so many changes.

Reply:
I couldn't to make From on new syn to work.
Maybe some features disabled?

Reply:
Probably. `from` should be the same as `new` but automatically using the current call site as span but maybe that has changed again in a newer version.

Reply:
```
 the trait `std::convert::From<&str>` is not implemented for `proc_macro2::Ident`
```

Reply:
https://github.com/alexcrichton/proc-macro2/issues/93

Reply:
`proc_macro2::Indent`?

Reply:
syn reexport Ident from proc_macro2 now

Reply:
Meh, this is all so weird.

Reply:
Okay, fine, let's go full on nuts with this. Will need proper testing though.

Reply:
bors try